rfc/mediachain-rfc-4.md

```diff
@@ -0,0 +1,450 @@
+# RFC 4: New Directions in Mediachain: Protocols and Architecture
+
+Status: WIP WIP WIP WIP
+
+Author: vyzo
+
+## Lessons from Phase I: It's not a Blockchain
+
+The blockchain is an attractive solution: the public ledger promotes
+open data and discovery, while also simplifying streaming for online
+processing.
+
+But it comes with significant costs:
+
+* Complexities of maintaining consensus in a peer to peer network.
+* PoS/PoW economics that don't quite work in our domain.
+* Inefficient querying, requires the entire blockchain for read operations.
+* Difficult to scale to large datasets with bulk and firehose ingestion, the blockchain becomes the bottleneck.
+* Ultimately trying to solve a problem we don't have: double spending.
+
+In reality, we don't need a singular linearly ordered view of the world.
+And we want to allow the system to grow to very large datasets (billions
+of objects) supporting bulk and firehose ingestion models.
+
+Fundamentally, what we need is a map of media identifiers to sets
+of statements that allows us to track and relate media objects.
+This has the hallmarks of a CRDT data structure, which allows a system to
+achieve eventually consistent state without the need for consensus. 
+
+So what we want to build is a distributed database that supports
+upserts, connecting statements to one or more domain specific identifiers.
```

Comment:
> jesse: how has thinking evolved re: identifiers?

[this probably deserves its own section]

We are choosing to move from opaque/mediachain-issued identifiers to well-known/domain-specific identifiers (let's call them WKIs here) (e.g. [moma:190935](http://www.moma.org/collection/works/190935?locale=en)) with the following motivations:

* it reduces burden of authoritative ID issuance (e.g. DOI) as well as authoritatively asserting canonical-instance relationships
* most works we are dealing with at this stage will already have an ID established by previous use (see above)
* mapping from a set of WKIs to a set of statements is an elegant way of granularly and nondestructively expressing [sameness/merges](https://github.com/mediachain/mediachain/issues/104), with potentially different "sameness" perspectives represented through namespace relationships (e.g. Clarifai can publish a namespace containing only merges based on visual similarity)
* WKI-based provider records can be reasonably implemented as distributed, freeing orgs from relying on indexers for simple mappings


Works that _are_ mediachain-first and require an identifier will receive one from a ticket server/blockchain implementation (TBD). This ID will be treated the same way as any other.